source/core/Array.cpp

```diff
@@ -1119,24 +1119,43 @@ VIREO_FUNCTION_SIGNATURET(ArrayMaxMinInternal, FindArrayMaxMinInstruction)
     IntIndex maxIndex = 0, minIndex = 0;
     AQBlock1* minValue = arrayIn->BeginAt(0);
     AQBlock1* maxValue = minValue;
-    if (len == 0)
+    if (len == 0) {
         maxIndex = minIndex = -1;
+    }
+    else {
+        if (arrayIn->ElementType()->IsFloat()) {
+            maxIndex = minIndex = -1;
+            if (arrayIn->ElementType()->TopAQSize() == sizeof(Double)) {
+                for (IntIndex i = 0; i < len; ++i) {
+                    minValue = arrayIn->BeginAt(i);
+                    if (!::isnan(*(Double*)minValue)) {
+                        maxIndex = minIndex = i;
+                        maxValue = minValue;
+                        break;
+                    }
+                }
+            }
+            else {
```

Comment:
It actually occurs to me that since this loop will stop the first time it finds a number that's not a NaN, it might be okay to combine this into one loop instead of two, because it will usually only do one iteration.

Reply:
I'm confused. Which two loops are you referring to?

Reply:
The separate loops for Double and Float.  It's an optimization we'd normally do to avoid the comparison each time through the loop, although in this case, the loop will almost never run so it might not be worth the code bloat.  So, this instead:

```
      if (arrayIn->ElementType()->IsFloat()) {
          maxIndex = minIndex = -1;
          bool isDouble = (arrayIn->ElementType()->TopAQSize() == sizeof(Double));
          for (IntIndex i = 0; i < len; ++i) {
              minValue = arrayIn->BeginAt(i);
              if (isDouble && !::isnan(*(Double*)minValue) || !isDouble &&  !::isnan(*(Single*)minValue) {
                  maxIndex = minIndex = i;
                  maxValue = minValue;
                  break;
              }
          }
      }
```

Reply:
Unless it's an array full of NaN's (I would think super rare), the for loop will usually break the first iteration, and not be any more expensive than having the if outside the loop.